### PR TITLE
Neon CLI updates

### PR DESCRIPTION
**Pull request by danieltprice**
Added proper upgrade instructions:
https://neon-next-git-dprice-cli-docs-updates-neondatabase.vercel.app/docs/reference/cli-install#upgrade

Mention --context_file option throughout. For example:
https://neon-next-git-dprice-cli-docs-updates-neondatabase.vercel.app/docs/reference/cli-auth#options

Added branch reset options:
https://neon-next-git-dprice-cli-docs-updates-neondatabase.vercel.app/docs/reference/cli-branches#reset

Added IP Allow options and examples:
https://neon-next-git-dprice-cli-docs-updates-neondatabase.vercel.app/docs/reference/cli-projects#update

### CODE REVIEW

File: content/docs/reference/cli-auth.md [hunk 1]
```diff
@@ -36,8 +36,12 @@ The authentication flow for the Neon CLI follows this order:
 - If the credentials file is not found, the Neon CLI initiates the `neonctl auth` web authentication process.
 </Admonition>
 
-### Options
+#### Options
 
-Only [global options](/docs/reference/neon-cli#global-options) apply.
+In addition to the Neon CLI [global options](/docs/reference/neon-cli#global-options), the `auth` subcommand supports this option:
```
Comment (duskpoet) [inline content/docs/reference/cli-auth.md:41]: it doesn't

Comment (duskpoet) [reply content/docs/reference/cli-auth.md:41]: I will fix the help message!

File: content/docs/reference/cli-me.md [hunk 2]
```diff
@@ -22,7 +22,11 @@ neonctl me
 
 ### Options
 
-Only [global options](/docs/reference/neon-cli#global-options) apply.
+In addition to the Neon CLI [global options](/docs/reference/neon-cli#global-options), the `me` subcommand supports this option:
+
```
Comment (duskpoet) [inline content/docs/reference/cli-me.md:26]: Same here, context is only for project/branch scoped commands

**Review by duskpoet — CHANGES_REQUESTED** [review]

File: content/docs/reference/cli-auth.md (hunk 1 above)
Comment (danieltprice) [reply content/docs/reference/cli-auth.md:41]: Thank you. Fixed.

File: content/docs/reference/cli-me.md (hunk 2 above)
Comment (danieltprice) [reply content/docs/reference/cli-me.md:26]: Fixed.

**Review by duskpoet — APPROVED** [review]

**Pull request merged**